.github/workflows/ocwm-issue-collector.yml

```diff
@@ -0,0 +1,86 @@
+name: Search for issues and PR labeled 'agenda' and add them to the agenda
+
+on:
+  schedule:
+    - cron: '0 0 * * 0'  # Runs every Sunday at midnight
+
+  repository_dispatch:
+    types: add-issues-ocwm
+
+jobs:
+  search_and_add:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install dependencies
+      run: npm install @octokit/core
+    - name: Adding Issues
+      uses: actions/github-script@v6
+      env:
+        PLACEHOLDER: '<!-- | [TOPIC] [IssuePRDiscussion] | [owner] | -->'
+        OWNER: ${{ vars.ORGANISATION }}
+        REPO: 'community'
+        REPO_NAMES: ${{ vars.REPOSITORIES }}
+        LABELS: ${{ vars.LABELS }}
+        MY_TOKEN: ${{ secrets.AUTH_TOKEN }}
+      with:
+        script: |
+          try {         
+            const repoNames = process.env.REPO_NAMES;
+            const labels = process.env.LABELS;
+            const placeholder = process.env.PLACEHOLDER;
+            console.log("Placeholder:" + placeholder);
+
+            const octokit = require('@octokit/core').Octokit  
+            const mygithub = new octokit({ auth: process.env.MY_TOKEN })
+
+            let repositories = repoNames.split(",")
+            let targetLabel = encodeURIComponent(labels.split(',')[0])
+            let appendLabel = encodeURIComponent(labels.split(',')[1])
+            console.log("Repositories:" + repositories);
+            for (let r = 0; r < repositories.length; r++) {
+              console.log(`GET /repos/${process.env.OWNER}/${repositories[r]}/issues?labels=${targetLabel}`);
+              console.log(`GET /repos/${process.env.OWNER}/${repositories[r]}/issues?labels=${appendLabel}&per_page=1`);
+
+              const { data: items2add } = await mygithub.request(`GET /repos/${{env.OWNER}}/${repositories[r]}/issues?labels=${targetLabel}`, {
+              });
+
+              console.log("Items to add:" + JSON.stringify(items2add));
+
+              const { data: workMeetings } = await mygithub.request(`GET /repos/${{env.OWNER}}/${{env.REPO}}/issues?labels=${appendLabel}&per_page=1`, {
+              })
+
+              console.log("workMeetings:" + JSON.stringify(workMeetings));
```

Comment:
This should be fetched before we enter the loop.  It's the same for every iteration (it doesn't depend on `r` at all).

By doing this, we can check to see if there are any open meetings _before_ we enter the loop.  If there are none, just return.  (As a follow-up to this PR, we may add the ability to generate a working meeting issue.)

Reply:
Agreed.